lib/Twig/Template.php

```diff
@@ -305,6 +305,33 @@ public function getBlocks()
     }
 
     /**
+     * Returns the template source code.
+     *
+     * @return string|null The template source code or null if it can't be retrieved
+     */
+    public function getSource()
+    {
+        $reflector = new \ReflectionClass($this);
+        $file = $reflector->getFileName();
+
+        if (!file_exists($file)) {
+            return;
+        }
+
+        $source = file($file);
+        array_splice($source, 0, $reflector->getEndLine());
```

Comment:
but would consume more memory, requiring a temporary duplication of the array...